stdlib/src/os/path/path.mojo

```diff
@@ -390,3 +391,154 @@ def split[PathLike: os.PathLike, //](path: PathLike) -> (String, String):
 #         paths_str.append(cur_path[].__fspath__())
 
 #     return join(path.__fspath__(), *paths_str)
+
+
+# ===----------------------------------------------------------------------=== #
+# expandvars
+# ===----------------------------------------------------------------------=== #
+
+
+fn _is_shell_special_variable(byte: Byte) -> Bool:
+    """Checks if `$` + `byte` identifies a special shell variable, such as `$@`.
+
+    Args:
+        byte: The byte to check.
+
+    Returns:
+        True if the byte is a special shell variable and False otherwise.
+    """
+    alias shell_variables = InlineArray[Int, 17](
+        ord("*"),
+        ord("#"),
+        ord("$"),
+        ord("@"),
+        ord("!"),
+        ord("?"),
+        ord("-"),
+        ord("0"),
+        ord("1"),
+        ord("2"),
+        ord("3"),
+        ord("4"),
+        ord("5"),
+        ord("6"),
+        ord("7"),
+        ord("8"),
+        ord("9"),
+    )
+    return int(byte) in shell_variables
```

Comment:
```suggestion
    alias `0` = Byte(ord("0"))
    alias `9` = Byte(ord("9"))
    alias shell_variables = SIMD[DType.uint8, 8](
        Byte(ord("!")),
        Byte(ord("!")),
        Byte(ord("#")),
        Byte(ord("$")),
        Byte(ord("*"),
        Byte(ord("-")),
        Byte(ord("?")),
        Byte(ord("@")),
    )
    return `0` <= byte <= `9` or byte in shell_variables
```

Reply:
I'm seeing some compilation issues using SIMD, so I'm going to leave it as it for now. `Byte(ord(...))` was adding visual noise IMO.